### ISSUE TITLE
Unset protobuf values in JSON serialization

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#1588]
I recently [submitted a PR](https://github.com/ga4gh/server/pull/1530) that reduced the cruft in transmitted JSON. I wanted to note for others that I noticed some unexpected behavior. When setting a boolean as false for serializing in a top level field of a message, it seems like the JSON serializer simply removes keys that have the default value. This make sense if you know you want to assume falsity when deserializing, but we don't make these assumptions explicit.

There are only a couple of places in the API that use the `bool` type https://github.com/ga4gh/schemas/search?utf8=%E2%9C%93&q=bool . Since it is an essentially unused and untested surface, it seems like this doesn't show in any tests. This is probably best solved by using `enum` in the schemas, as this makes the default value explicit.

**Comment by dcolligan** [comment]
Huh, that's confusing behavior.  And the workaround is a good one, if the problem is limited to just bools.  (There is a school of thought that I am somewhat partial to that says that bools should never be used in the first place, but enums always used instead, since they are inevitably generalized and the subsequent refactor becomes easier if one doesn't need to switch types.)

It doesn't do the same for other sink values, like empty strings and 0-value ints, I hope?

**Comment by david4096** [comment]
Yeah, I think it was intentional on Google's part to still support `optional` fields for basic types. If you go down to where it talks about optional [here](https://developers.google.com/protocol-buffers/docs/pythontutorial) it states:

> For simple types, you can specify your own default value, as we've done for the phone number type in the example. Otherwise, a system default is used: zero for numeric types, the empty string for strings, false for bools. For embedded messages, the default value is always the "default instance" or "prototype" of the message, which has none of its fields set. Calling the accessor to get the value of an optional (or required) field which has not been explicitly set always returns that field's default value. 

Try this:

```
In [27]: from ga4gh.schemas import protocol

In [28]: variant = protocol.Variant()

In [29]: variant.reference_bases = ""

In [30]: print(variant)


In [31]: variant.alternate_bases.append("")

In [32]: print(variant)
alternate_bases: ""
```

This is working for us, although it seems like the kind of thing that could keep a programmer up at night. When you're designing a type system, this is quite confusing, but when you're designing an interchange format one might be more accepting of that kind of assumption. Protobuf doesn't "represent" data it just transfers it. As far as I can tell, it's not possible to serialize a blank string, it is assumed that if someone goes to look for the value of the tag and it is not there, it is assumed to be `""`.

If you give up that Protocol Buffers is a data modeling language, and that it is optimized for making interchange protocols, you can see how the rest of their stack would take shape. Applications that reflect on the proto are able to impute values instead of putting them on the wire when they are the default value.

The problem then becomes missing values, since a default is assumed. How do I say if I actually mean, "I don't know this value". This issue came up with the `Genotype` list in calls, which now can represent missing values using a `oneof` that includes nullity.

I posted this here so developers at the server level might observe the behavior and not make incorrect assumptions about what protocol buffers provides. With @dcolligan's comment we might make a PR at the schemas level that refactors `bool` to `enum` in those few places.
